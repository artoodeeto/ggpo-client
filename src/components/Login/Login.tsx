import React from 'react';
// import LoginStyles from './App.module.scss';

function testSubmit(event: any) {
  event.preventDefault();
  console.log(event);
}

class Login extends React.Component {
  render() {
    return (
      <div>
        <div>
          <form onSubmit={testSubmit}>
            <label htmlFor="email">
              Username:
              <input placeholder="youremail@address.com" type="text" name="email" id="" required />
            </label>

            <label htmlFor="password">
              Password:
              <input placeholder="password" type="text" name="email" id="" required />
            </label>

            <input type="submit" value="Login" />
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
