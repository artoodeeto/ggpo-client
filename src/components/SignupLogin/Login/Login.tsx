import React, { FormEvent, useState, FC } from 'react';
import { connect } from 'react-redux';
// import LoginStyles from './App.module.scss';
import { ILoginSignUpFormParams } from 'interfaces/session';
import { showLoading } from 'store/session/Selectors';
import { logMeIn } from 'store/session/Actions';
import { RootState } from 'store/root/root_reducer';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

type LoginProps = {
  toShowLoading: boolean;
  onLogin: (userLogin: ILoginSignUpFormParams) => void;
};

const onSubmitLogin = (
  event: FormEvent,
  email: string,
  password: string,
  onLogin: (userLogin: ILoginSignUpFormParams) => void
): void => {
  event.preventDefault();

  const formSignup: ILoginSignUpFormParams = {
    email,
    password
  };
  onLogin(formSignup);
};

const Login: FC<LoginProps> = ({ toShowLoading, onLogin }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <div>
      <div>
        <div>{toShowLoading ? 'LOGGING EN' : ''}</div>
        <form onSubmit={(e) => onSubmitLogin(e, email, password, onLogin)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Login" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    toShowLoading: showLoading(state)
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    onLogin: (userLogin: ILoginSignUpFormParams) => dispatch(logMeIn(userLogin))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
