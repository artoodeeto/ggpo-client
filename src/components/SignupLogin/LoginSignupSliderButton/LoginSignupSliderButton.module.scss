.SliderBtn {
  font-size: 1em;
}
