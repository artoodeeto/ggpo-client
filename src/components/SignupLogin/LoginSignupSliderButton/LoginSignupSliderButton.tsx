import React, { FC, Fragment } from 'react';
import styles from './LoginSignupSliderButton.module.scss';

type LoginSignupSliderButtonProps = {
  toShow: (val: boolean) => void;
};

export const LoginSignupSliderButton: FC<LoginSignupSliderButtonProps> = ({ toShow }) => {
  return (
    <Fragment>
      <button className={styles.SliderBtn} type="button" onClick={() => toShow(true)}>
        Already a User?
      </button>
      <button className={styles.SliderBtn} type="button" onClick={() => toShow(false)}>
        New User
      </button>
    </Fragment>
  );
};
