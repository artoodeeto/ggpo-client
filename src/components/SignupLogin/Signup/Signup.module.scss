@import '../../../assets/media-queries.scss';

.Signup__Form {
  display: flex;
  flex-direction: column;
  align-items: center;

  > input {
    font-size: 1em;
    margin: 0.2em;
  }
}
