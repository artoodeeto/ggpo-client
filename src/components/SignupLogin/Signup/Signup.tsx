import React, { useState, useEffect, FormEvent, FC } from 'react';
import * as userSelectors from 'store/user/Selectors';
import * as sessionSelectors from 'store/session/Selectors';
import * as sessionActions from 'store/session/Actions';
import { connect } from 'react-redux';
import { IState } from 'interfaces/stateInterface';
import { ILoginSignUpFormParams } from 'interfaces/session';

const onSubmitSignup = (
  event: FormEvent,
  email: string,
  username: string,
  password: string,
  onSignup: Function
): void => {
  event.preventDefault();

  const formSignup: ILoginSignUpFormParams = {
    username,
    email,
    password
  };

  onSignup(formSignup);
};

export const Signup: FC = (props: any) => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {}, [username, email, password]);

  return (
    <div>
      <div>
        <form onSubmit={(e) => onSubmitSignup(e, email, username, password, props.onSignup)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="username">
            Username:
            <input
              onChange={(e) => setUsername(e.target.value)}
              value={username}
              placeholder="user name"
              type="text"
              name="username"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Sigup" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: IState) => {
  return {
    sign: 'up',
    isAuthenticated: sessionSelectors.isUserAuthorized(state),
    userInfo: userSelectors.userInfo(state)
  };
};

const mapDispatchToProps = (dispatch: Function) => {
  return {
    onSignup: (userSignup: ILoginSignUpFormParams) => dispatch(sessionActions.signMeUp(userSignup))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Signup);
