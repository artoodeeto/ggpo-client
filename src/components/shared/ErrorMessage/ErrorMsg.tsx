import React, { FC } from 'react';
import { connect } from 'react-redux';
import { RootState } from 'store/root/root_reducer';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { authErrorsResponse } from 'store/session/Selectors';
import { ILoginSignupFailed } from 'interfaces/session';

type ErrorMsgProps = {
  errorMsg: ILoginSignupFailed;
};

const ErrorMsg: FC<ErrorMsgProps> = ({ errorMsg }) => {
  const errMsg = errorMsg.error.errors[0];
  const errKeys = Object.keys(errMsg);

  return (
    <div>
      <ul>
        {errKeys.map((errKey, i) => (
          <li key={i}>{errMsg[errKey]}</li>
        ))}
      </ul>
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    errorMsg: authErrorsResponse(state)
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(ErrorMsg);
