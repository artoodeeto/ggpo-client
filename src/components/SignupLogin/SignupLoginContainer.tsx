import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import ErrorMsg from 'components/shared/ErrorMessage/ErrorMsg';
import { RootState } from 'store/root/root_reducer';
import { hasErrorOnSignupOrLogin, isUserAuthorized } from 'store/session/Selectors';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { Signup } from './Signup/Signup';
import { Login } from './Login/Login';
import styles from './SignupLoginContainer.module.scss';
import { LoginSignupSliderButton } from './LoginSignupSliderButton/LoginSignupSliderButton';

interface SignupLoginContainerProps {
  hasErrors: boolean;
  isAuthenticated: boolean;
}

interface SignupLoginContainerState {
  toLoginOrSignup: boolean;
}

// background here should show some video, maybe the TI championship games
// TODO: sign up and login button should show loading GIF on the their buttons
class SignupLoginContainer extends Component<SignupLoginContainerProps, SignupLoginContainerState> {
  constructor(props: SignupLoginContainerProps) {
    super(props);
    this.state = {
      toLoginOrSignup: true
    };
  }

  toLogInOrSignUp(toShow: boolean): void {
    this.setState({ toLoginOrSignup: toShow });
  }

  render() {
    return (
      <Fragment>
        {this.props.hasErrors ? (
          <div>
            <ErrorMsg />
          </div>
        ) : null}

        {this.props.isAuthenticated ? (
          <Redirect to="/feed" />
        ) : (
          <div className={styles.SignupLoginContainer}>
            <div className={styles.SignupLoginContainer__SliderBtnContainer}>
              <LoginSignupSliderButton toShow={(val: boolean) => this.toLogInOrSignUp(val)} />
            </div>
            <div className={styles.SignupLoginContainer__Form_Container}>
              {this.state.toLoginOrSignup ? (
                <Login isLoggingInOrSigningUp={this.state.toLoginOrSignup} />
              ) : (
                <Signup isLoggingInOrSigningUp={this.state.toLoginOrSignup} />
              )}
            </div>
          </div>
        )}
      </Fragment>
    );
  }
}

const mapStateToProps = (state: RootState) => {
  return {
    hasErrors: hasErrorOnSignupOrLogin(state),
    isAuthenticated: isUserAuthorized(state)
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(SignupLoginContainer);
