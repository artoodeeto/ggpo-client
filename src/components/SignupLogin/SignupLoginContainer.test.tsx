/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';
import ErrorMsg from 'components/shared/ErrorMessage/ErrorMsg';
import { Redirect } from 'react-router-dom';
import { shallowToJson } from 'enzyme-to-json';
import { SignupLoginContainer } from './SignupLoginContainer';
import { Login } from './Login/Login';
import { Signup } from './Signup/Signup';
import { LoginSignupSliderButton } from './LoginSignupSliderButton/LoginSignupSliderButton';

describe('<SignupLoginContainer/>', () => {
  describe('SignupLoginContainer', () => {
    it('container should match snapshots if user is logging in', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
      // expect(wrapper.instance().props).toContainKeys(['hasErrors', 'isAuthenticated']);
      // expect(wrapper.state()).toContainKey('toLoginOrSignup');
    });

    it('container should match snapshots if user is signing up', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      wrapper.setState({
        toLoginOrSignup: false
      });
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('SignupLoginContainer should contain props and state', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      // expect(wrapper.instance().props).toContainKeys(['hasErrors', 'isAuthenticated']);
      // expect(wrapper.state()).toContainKey('toLoginOrSignup');
    });

    it('should have classnames', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      expect(wrapper.find('.SignupLoginContainer')).toHaveClassName('SignupLoginContainer');
      // expect(wrapper.find('.SignupLoginContainer__SliderBtnContainer')).toHaveClassName(
      //   'SignupLoginContainer__SliderBtnContainer'
      // );
      // expect(wrapper.find('.SignupLoginContainer__Form_Container')).toHaveClassName(
      //   'SignupLoginContainer__Form_Container'
      // );
    });
  });

  describe('USER have errors logging in', () => {
    it('should show ErrorMsg component', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors isAuthenticated={false} />);
      const errComp = wrapper.find(ErrorMsg);
      expect(errComp.exists()).toBeTrue();
    });
  });

  describe('USER Redirected if already login', () => {
    it('should show Redirect component with  "to" as props and "/feed" as value', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated />);
      const redirect = wrapper.find(Redirect);
      expect(redirect.exists()).toBeTrue();
      expect(redirect.props().to).toEqual('/feed');
    });
  });

  describe('Slider button should show all the time', () => {
    it('shows slider button component', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      // const sliderBtn = wrapper.find(LoginSignupSliderButton);
      // expect(sliderBtn.props()).toContainKey('toShow');
      // expect(sliderBtn.exists()).toBeTrue();
    });
  });

  describe('USER SIGNING UP', () => {
    it('should show Signup component', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      // const sliderBtn = wrapper.find(LoginSignupSliderButton);
      // const spy = jest.spyOn(SignupLoginContainer.prototype, 'toLogInOrSignUp');

      // sliderBtn.prop('toShow')(false);
      // sliderBtn.prop('toShow')(false);
      // expect(spy).toHaveBeenCalledTimes(2);
      const signupComp = wrapper.find(Signup);
      expect(signupComp.exists()).toBeTrue();
      expect(signupComp.props()).toContainKeys(['isLoggingInOrSigningUp']);
      // spy.mockRestore();
    });
  });

  describe('USER LOGGING IN', () => {
    it('should show Login component', () => {
      const wrapper = shallow(<SignupLoginContainer hasErrors={false} isAuthenticated={false} />);
      const loginComp = wrapper.find(Login);
      expect(loginComp.props()).toContainKeys(['isLoggingInOrSigningUp']);
      expect(loginComp.exists()).toBeTrue();
    });
  });
});
