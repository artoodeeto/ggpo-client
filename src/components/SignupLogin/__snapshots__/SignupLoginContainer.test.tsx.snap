// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupLoginContainer/> SignupLoginContainer container should match snapshots if user is logging in 1`] = `
<Fragment>
  <div
    className="SignupLoginContainer"
  >
    <Login
      isLoggingInOrSigningUp={true}
    />
    <span>
      or
    </span>
    <Signup
      isLoggingInOrSigningUp={false}
    />
  </div>
</Fragment>
`;

exports[`<SignupLoginContainer/> SignupLoginContainer container should match snapshots if user is signing up 1`] = `
<Fragment>
  <div
    className="SignupLoginContainer"
  >
    <Login
      isLoggingInOrSigningUp={true}
    />
    <span>
      or
    </span>
    <Signup
      isLoggingInOrSigningUp={false}
    />
  </div>
</Fragment>
`;
